src/dynarec/rv64/dynarec_rv64_private.h

```diff
@@ -18,20 +18,27 @@ typedef struct instsize_s instsize_t;
 #define EXT_CACHE_SS     5
 #define EXT_CACHE_SD     6
 #define EXT_CACHE_SCR    7
+#define EXT_CACHE_XMMW   8
+#define EXT_CACHE_XMMR   9
+
 typedef union ext_cache_s {
     int8_t           v;
     struct {
         uint8_t t:4;   // reg type
         uint8_t n:4;   // reg number
     };
 } ext_cache_t;
+
 typedef union sse_cache_s {
-    int8_t      v;
+    int8_t v;
     struct {
-        uint8_t     reg:7;
-        uint8_t     single:1;
+        uint8_t reg:5;
```

Comment:
If there is only 16 regs then we have a problem, and cannot use fixed register position as for arm64, because that would leave 0 scratch register!
We need sommething more dynamic, like YMM handling on ARM64, because we need more than 16regs!

Reply:
wait... it stores the native register number, but the sse register number, then there is 32 of them, and we need a uint16_t.

Reply:
I reverted the change, sorry.

Reply:
reverted partly... 5bits should be enough yes.

Reply:
5 bits are not enough I think, because we use -1 to indicate the register is forgotten, so there needs at least 1 bit free to not conflict with -1.

Reply:
sorry not sure I followed, but what you suggested on "revert partly" is this?

```
       uint16_t reg:5;
        uint16_t vector:1;
        uint16_t single:1;
        uint16_t write:1;
        uint16_t isfree:1;
        uint16_t unused:7;
```

Reply:
Yes. And this `isfree` need to be set to 0 when used...